Mathlib/FieldTheory/JacobsonNoether.lean

```diff
@@ -0,0 +1,258 @@
+/-
+Copyright (c) 2024 Huanyu Zheng, Wanyi He, Sihan Wu, Yi Yuan, Filippo A. E. Nuccio.
+All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Huanyu Zheng, Sihan Wu, Wanyi He, Yi Yuan, Filippo A. E. Nuccio.
+-/
+import Mathlib.FieldTheory.Separable
+import Mathlib.Algebra.CharP.Subring
+import Mathlib.Algebra.CharP.LinearMaps
+
+/-!
+# The Jacobson-Noether theorem
+
+This file contains a proof of the Jacobson-Noether theorem and some auxiliary lemmas.
+Here we discuss different cases of characteristics of
+the noncommutative division algebra `D` with the center `k`.
+
+## Main Results
+
+- `Jacobson_Noether` : Let `D` be a noncommutative division algebra with the center `k`.
+  If `D` is algebraic over `k`,
+  then there exists an element of `D \ k` which is separable over `k`.
+
+## Notations
+
+- `D` is a noncommutative division algebra
+- `k` is the center of `D`
+
+## Implementation Notes
+
+## Reference
+* <https://ysharifi.wordpress.com/2011/09/30/the-jacobson-noether-theorem/>
+-/
+
+-- *Filippo* This should probably be moved to another file.
+/-- a⁻¹ * d ^ s * a = (a⁻¹ * d * a) ^ s when a is not zero -/
+lemma conj_nonComm_Algebra {D : Type*} [DivisionRing D] (s : ℕ) (a d : D) (ha : a ≠ 0) :
+    a⁻¹ * d ^ s * a = (a⁻¹ * d * a) ^ s := by
+  let u : Dˣ := ⟨a, a⁻¹, mul_inv_cancel₀ ha, inv_mul_cancel₀ ha⟩
+  exact (Units.conj_pow' u d s).symm
+
+namespace JacobsonNoether
+
+variable {D : Type*} [DivisionRing D]
+
+local notation3 "k" => (Subring.center D)
+
+instance : Algebra (Subring.center D) D := Algebra.ofModule smul_mul_assoc mul_smul_comm
+
+private def f (a : D) : D →ₗ[k] D := {
+  toFun := fun x ↦ a * x
+  map_add' := fun x y ↦ LeftDistribClass.left_distrib a x y
+  map_smul' := fun m x ↦ by simp only [mul_smul_comm, RingHom.id_apply]
+}
```

Comment:
Maybe you can use [LinearMap.mulLeft](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Algebra/Bilinear.html#LinearMap.mulLeft) for `f`, [LinearMap.mulRight](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Algebra/Bilinear.html#LinearMap.mulRight) for `g`, [LinearMap.commute_mulLeft_right](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Algebra/Bilinear.html#LinearMap.commute_mulLeft_right) for they commutes, etc. etc.

Reply:
good idea, implemented, thanks!